src/main/java/io/kokuwa/maven/helm/DependencyBuildMojo.java

```diff
@@ -28,6 +41,180 @@ public class DependencyBuildMojo extends AbstractHelmMojo {
+	 * "helm.overwriteDependencyVersion" and "helm.overwriteDependencyRepository". This is helpful for deploying charts 
+	 * with intra repository dependencies, while still being able to use local path dependencies for development builds.
+	 *
+	 * Example usage:
+	 * 	For development: mvn clean install
+	 * 	For deployment: mvn clean deploy -Dhelm.overwriteLocalDependencies=true
+	 * @since 6.9.1
+	 */
+	@Parameter(property = "helm.overwriteLocalDependencies", defaultValue = "false")
+	private boolean overwriteLocalDependencies;
+
+	/**
+	 * Value used to overwrite a local path chart's version within a chart's dependencies. The property 
+	 * "helm.overwriteLocalDependencies" must be set to <code>true</code> for this to apply.
+	 *
+	 * @since 6.9.1
+	 */
+	@Parameter(property = "helm.overwriteDependencyVersion")
+	private String overwriteDependencyVersion;
+
+	/**
+	 * Value used to overwrite a local path chart's repository within a chart's dependencies. The property 
+	 * "helm.overwriteLocalDependencies" must be set to <code>true</code> for this to apply.
+	 *
+	 * @since 6.9.1
+	 */
+	@Parameter(property = "helm.overwriteDependencyRepository")
+	private String overwriteDependencyRepository;
+
+	/**
+	 * Read in the dependencies from the Chart.yaml into the {@link Dependencies} POJO
+	 * @param chartFile Chart.yaml file to read
+	 * @return {@link ArrayList}<{@link Dependency}>
+	 * @throws MojoExecutionException
+	 */
+	private ArrayList<Dependency> getDependencies(File chartFile) throws MojoExecutionException {
+		ObjectMapper yamlMapper = new YAMLMapper();
+		yamlMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+		Dependencies dependencies;
+		try {
+			dependencies = yamlMapper.readValue(chartFile, Dependencies.class);
+		} catch (IOException e) {
+			throw new MojoExecutionException("Unable to read chart dependencies from " + chartFile, e);
+		}
+
+		return dependencies.getDependencies();
+	}
+
+	/**
+	 * Updates the {@link Dependency} with the new overwritten values
+	 * @param dependencies {@Link ArrayList}<{@link Dependency}>
+	 */
+	private void updateDependencies(ArrayList<Dependency> dependencies) {
+		for (Dependency dependency : dependencies) {
+			if (dependency.getRepository() != null && dependency.getRepository().contains("file://")) {
+				dependency.setRepository(overwriteDependencyRepository);
+				dependency.setVersion(overwriteDependencyVersion);
+				dependency.setOverwrite(true);
+			}
+		}
+	}
+
+	/**
+	 * Overwrite the dependencies with their new values to the Chart.yaml 
+	 * @param dependencies {@link ArrayList}<{@link Dependency}>
+	 * @param chartFile Chart.yaml file to read
+	 * @param chartDirectory Directory to write the new Chart.yaml to
+	 * @throws MojoExecutionException
+	 */
+	private void overwriteDependencies(ArrayList<Dependency> dependencies, File chartFile, Path chartDirectory) 
```

Comment:
Decided offline to pull this logic into a new Chart updater class so the logic is both compartmentalized and we can more easily add unit tests to ensure we're covering the cases we expect.